src/NuGet.Services.AzureSearch/HijackDocumentBuilder.cs

```diff
@@ -0,0 +1,89 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using NuGet.Protocol.Catalog;
+using NuGet.Services.Entities;
+
+namespace NuGet.Services.AzureSearch
+{
+    public class HijackDocumentBuilder : IHijackDocumentBuilder
+    {
+        public KeyedDocument Keyed(
+            string packageId,
+            string normalizedVersion)
+        {
+            var document = new KeyedDocument();
+
+            PopulateKey(document, packageId, normalizedVersion);
+
+            return document;
+        }
+
+        public HijackDocument.Latest Latest(
+            string packageId,
+            string normalizedVersion,
+            HijackDocumentChanges changes)
+        {
+            var document = new HijackDocument.Latest();
+
+            PopulateLatest(document, packageId, normalizedVersion, changes);
+
+            return document;
+        }
+
+        public HijackDocument.Full Full(
+            string normalizedVersion,
+            HijackDocumentChanges changes,
+            PackageDetailsCatalogLeaf leaf)
+        {
+            var document = new HijackDocument.Full();
+
+            PopulateFull(document, leaf.PackageId, normalizedVersion, changes, leaf.Title);
```

Comment:
>PopulateFull [](start = 12, length = 12)

Why doesn't `PopulateFull` call `PopulateMetadata`? It seems like "full" should include the metadata

Reply:
The parameters of `PopulateMetadata` vary for catalog2azuresearch vs. db2azuresearch. This is an internal method that is meant to sent the fields that have the same parameters between these two cases.

Reply:
Ah right. Ignore my comment